### ISSUE TITLE
Rules might run even if configuration errors has been already detected

### ISSUE THREAD

**Issue by chavacava** [mgechev/revive#1182]
The current mechanism for aborting the execution of rules when an error is detected in the configuration makes possible that rules can apply with incomplete/erroneous configurations until the main process exits.

For example, if we execute revive with the following `deleteme.toml` configuration
```toml
ignoreGeneratedHeader = false
severity = "warning"
confidence = 0.8
errorCode = 0
warningCode = 0

[rule.add-constant]
arguments = ["lint:ignore"]
```  

(with the rule being modified to print some traces) we can expect (and get) the following output:
```
>>>> configuring rule to lint file cli/main.go
>>>> (err in conf) aborting linting file cli/main.go
invalid argument to the add-constant rule, expecting a k,v map. Got string
```
the output shows:
1. a goroutine executes rule.Apply (on `cli/main.go`) and tries to configure the rule 
2. the configuration fails thus the Apply exits informing an error
3. (finally) the control of the program exits because the goroutine returned an error

but we could also get
```
>>>> configuring rule to lint file cli/main.go
>>>> (err in conf) aborting linting file cli/main.go
>>>> linting file main.go
>>>> linting file cli/main_test.go
>>>> linting file config/config_test.go
invalid argument to the add-constant rule, expecting a k,v map. Got string
```
the output shows:
1. a goroutine executes rule.Apply (on `cli/main.go`) and tries to configure the rule 
2. the configuration fails thus the Apply exits informing an error
3. another goroutine applies the rule to `main.go`... (no error in conf detected because sync.Once)
4. yet another goroutine applies the rule to `cli/main_test.go`... (no error in conf detected because sync.Once)  
5. yet another goroutine applies the rule to `config/config_test.go`... (no error in conf detected because sync.Once) 
6. (finally) the control of the program exits because the first goroutine returned an error

Notice that in 3,4, and 5 the rule is being applied without a complete/correct configuration, then the rule's behavior is undefined. 
Luckily, this problem is not observable by the user. That is because: a) the main control exits before the bad configured rule applications can do anything, and/or b) the formatters do not process failures on the fly but at the end of the analysis. 

Notice that the "randomness" of the CPU allocation to goroutines produces things like:
```
>>>> configuring rule to lint file cli/main.go
>>>> (err in conf) aborting linting file cli/main.go
>>>> linting file internal/typeparams/typeparams_go118.go
>>>> linting file config/config_test.go
>>>> linting file internal/ifelse/branch_kind.go
>>>> linting file main.go
>>>> linting file cli/main_test.go
>>>> linting file config/config.go
>>>> linting file revivelib/extra_rule.go
>>>> linting file logging/logger.go
>>>> linting file revivelib/core.go
>>>> linting file formatter/severity.go
>>>> linting file internal/astutils/ast_utils.go
>>>> linting file internal/ifelse/chain.go
>>>> linting file internal/ifelse/doc.go
>>>> linting file internal/ifelse/func.go
>>>> linting file formatter/friendly.go
>>>> linting file internal/ifelse/target.go
>>>> linting file internal/ifelse/args.go
>>>> linting file internal/ifelse/branch.go
>>>> linting file formatter/plain.go
>>>> linting file formatter/unix.go
>>>> linting file formatter/sarif.go
>>>> linting file formatter/doc.go
>>>> linting file formatter/stylish.go
>>>> linting file formatter/checkstyle.go
>>>> linting file internal/ifelse/rule.go
>>>> linting file formatter/default.go
>>>> linting file revivelib/pattern.go
>>>> linting file formatter/json.go
>>>> linting file formatter/ndjson.go
>>>> linting file internal/typeparams/typeparams.go
invalid argument to the add-constant rule, expecting a k,v map. Got string
```

cc: @ccoVeille

**Comment by ccoVeille** [comment]
Thanks for the extra information to replicate.

I will try a few things
